tests/java/org/pantsbuild/tools/junit/impl/ConsoleRunnerImplTest.java

```diff
@@ -409,35 +412,35 @@ private String getTestLogContents(Class testClass, String extension) {
     }
   }
 
-  private <T> Matcher<Iterable<? super T>> hasExactlyOneOf(final Matcher<? super T> elemMatcher) {
+  private <T> Matcher<Iterable<T>> hasExactlyOneOf(final Matcher<T> elemMatcher) {
     return new ExactlyOneOf<T>(elemMatcher);
   }
 
-  private static class ExactlyOneOf<T> extends TypeSafeDiagnosingMatcher<Iterable<? super T>> {
+  private static class ExactlyOneOf<T> extends TypeSafeDiagnosingMatcher<Iterable<T>> {
     private final Matcher<? super T> elemMatcher;
 
     public ExactlyOneOf(Matcher<? super T> elemMatcher) {
       this.elemMatcher = elemMatcher;
     }
 
     @Override
-    protected boolean matchesSafely(Iterable<? super T> collection,
+    protected boolean matchesSafely(Iterable<T> iterable,
                                     Description mismatchDescription) {
-      boolean foundOne = false;
-      String s;
-      for (Object t : collection) {
-        if (elemMatcher.matches(t)) {
-          if (foundOne) {
-            mismatchDescription.appendText("found more than one");
-            return false;
-          }
-          foundOne = true;
-        }
+      List<T> filtered = StreamSupport.stream(iterable.spliterator(), false)
```

Comment:
Since `StreamSupport` is java 1.8+  you should just be able to pass `elemMatcher::matches` to `filter`.

Reply:
passing that will break running jarjar on this if it decomposes into a lambda. I'd assumed that method references like that do use the lambda API.

Reply:
Really? Our jarjar uses asm 7 and that supports up through java 11. ~~Is there some petard in our jarjar compiling / packaging / publishing we can an-hoist?~~

OK, our petard is here: https://github.com/pantsbuild/jarjar/blob/57845dc73d3e2c9b916ae4a788cfa12114fd7df1/pants.ini#L20-L25

Clearly that should not block this.

Reply:
Specifically this issue. https://github.com/pantsbuild/jarjar/issues/51

Reply:
now I want to see if I can exercise it with a test in jarjar.